enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/AuthProcedures.java

```diff
@@ -362,12 +276,13 @@ private StandardEnterpriseAuthSubject ensureAdminAuthSubject()
         return enterpriseAuthSubject;
     }
 
-    private StandardEnterpriseAuthSubject ensureSelfOrAdminAuthSubject( String username ) throws InvalidArgumentsException
+    private StandardEnterpriseAuthSubject ensureSelfOrAdminAuthSubject( String username )
+            throws InvalidArgumentsException
     {
         StandardEnterpriseAuthSubject subject = StandardEnterpriseAuthSubject.castOrFail( authSubject );
         subject.getUserManager().getUser( username );
 
-        if ( subject.isAdmin() || subject.doesUsernameMatch( username ) )
+        if ( subject.isAdmin() || subject.hasUsername( username ) )
```

Comment:
This should be reordered as Olivia said above. I guess
`if ( subject.hasUsername( username ) || ( subject.isAdmin() && subject.getUserManager().getUser( username ) ) )` would cut it